extensions/markdown-language-features/src/languageFeatures/copyFiles/copyPasteLinks.ts

```diff
@@ -43,6 +40,86 @@ class PasteLinkEditProvider implements vscode.DocumentPasteEditProvider {
 	}
 }
 
+// function checkURL(item: vscode.DataTransferItem | undefined) {
+// 	try {
+// 		const url = new URL(item?.value);
+// 		return externalUriSchemes.includes(url.protocol.slice(0, -1));
+// 	} catch (error) {
+// 		return false;
+// 	}
+// }
+
+const CONTROL_CODES = '\\u0000-\\u0020\\u007f-\\u009f';
+const WEB_LINK_REGEX = new RegExp('(?:[a-zA-Z][a-zA-Z0-9+.-]{2,}:\\/\\/|data:|www\\.)[^\\s' + CONTROL_CODES + '"]{2,}[^\\s' + CONTROL_CODES + '"\')}\\],:;.!?]', 'ug');
+const WIN_ABSOLUTE_PATH = /(?<=^|\s)(?:[a-zA-Z]:(?:(?:\\|\/)[\w\.-]*)+)/;
```

Comment:
At this time let's focus on full uris (`scheme:...`) instead of file system paths too. This should help simplify the code